server/src/main/java/org/elasticsearch/search/suggest/completion/context/GeoContextMapping.java

```diff
@@ -200,18 +201,29 @@ protected XContentBuilder toInnerXContent(XContentBuilder builder, Params params
                             geohashes.add(stringEncode(spare.getLon(), spare.getLat(), precision));
                         }
                     }
+                } else {
+                    // Does this object exist? If it does and we didn't find what we need - we need to warn user
+                    for (IndexableField field : document.getFields()) {
+                        if (field.name().startsWith(fieldName + ".")) {
```

Comment:
I know the "dots in field names" discussion has been a long running one. Do we not yet have a more general/graceful way of throwing these exceptions? This is more of a question out of my own curiosity and not intended to hold up the PR.

Reply:
I am not sure we should do that, can we restrict the geo context to `geo` fields and validate this assumption from the mapping (`context.mapperService`) directly ? If the field has the correct type in the mapping we don't have to check why it's missing.

Reply:
We can definitely remove that and clean this up even more, but this will be much more breaking change, since today you can have this sudo geo object and it works, if we will remove this in 7, then working indices with this, created in 6 are going to break. I am not sure we can do that. We will probably have to disable it for indices in 7 but I don't think we can fully remove it until 8.